README.md

```diff
@@ -0,0 +1,167 @@
+
+ // Now register the middleware with our router, indicating that we
+ // want to use the parser for JSON data.
+ router.use(bodyParser.json());
+```
+
+Now that we've got the body-parser in place, we are finally in a position to submit our message to the server. Excited? Good, I hope so. Let's see if you can get those tests to pass by yourself. To get you'll started here are a few hints and things to consider:
+
+1. Now that our body parser middleware is in place, you can access the data you pass to the server on the request object's body parameter: `request.body`.
+2. Note that unlike our Hello World server, the data we are sending back and forth here is JSON data. That's what our bodyParser is parsing. As such, we'll want to set the headers appropriately.
+3. How will we represent our messages on the server? Each of our messages needs an id, plus a text message. Maybe it's a good idea to create a javascript class? It's also possible that this would "over engineer" the problem. What do you think?
+
+## Retreiving Messages
+
+Okay, so now you've got aserver that's able to accept a POST request to `/message` and save the supplied message in memory. Wonderful. Now let's teach our server to retrieve messages..
+
+First, let's tackle the use-case where we just want to get all the messages. We'll use the route `/messages`. You should be able to do this by yourself. When you're done with that we'll move onto the sslightly more complicated scenario in which we'd like the server to return a specific message specified by its id. As you are working on this part, remeber to set the response headers correctly. You will also need to tranform your data into a JSON string before sending it to the server. Google a bit. You should be able to discover how to do this.
+
+## Retrieving a Specific Message
+
+Okay, this step is a bit more challenging. Again, we have to deal with some server setup so that we can easily extract the id from the request. First, let's think about how we will indicate to the server which message we want. We'll use a GET request for this route, as we are attempting to get rather than send (or post) data. We do, however, need to send the id of the message that we want. One elegant way of doing this is to simply attach the id to the end of the url path, like so:
+
+`/message/1`
+
+That's nice an concise, no? Now let's imagine how that form could be generalized. Something like this maybe:
+
+`/message/:id`
+
+Yep that makes sense, and in fact it's a common convention used by many node modules and other web libraries and frameworks. In order to get our server to make sense of this convention, however, we're going to have to modify the setup options for our router module. We need to activate the `mergeParams` functionality by sending some options into our router's constructor method when we instantiate it:
+
+```javascript
+const router = new Router({ mergeParams: true });
+```
+
+Now we are ready to define our route:
```

Comment:
This is great! Students will be generally familiar with path params from their work in Sinatra/Rails, but I think it will be new to them to have to "turn it on" -- I think the way you've presented this concept will really help it "click" for students.